src/Elf.cpp

```diff
@@ -444,7 +449,39 @@ Object::section_iterator Object::find_section(const std::string &name) {
     }
     return sections_end();
 }
-
+std::string Section::section_type_string() const {
+    switch(type) {
+    case SHT_NULL: return "SHT_NULL";
+    case SHT_PROGBITS: return "SHT_PROGBITS";
+    case SHT_SYMTAB: return "SHT_SYMTAB";
+    case SHT_STRTAB: return "SHT_STRTAB";
+    case SHT_RELA: return "SHT_RELA";
+    case SHT_HASH: return "SHT_HASH";
+    case SHT_DYNAMIC: return "SHT_DYNAMIC";
+    case SHT_NOTE: return "SHT_NOTE";
+    case SHT_NOBITS: return "SHT_NOBITS";
+    case SHT_REL: return "SHT_REL";
+    case SHT_SHLIB: return "SHT_SHLIB";
+    case SHT_DYNSYM: return "SHT_DYNSYM";
+    case SHT_LOPROC: return "SHT_LOPROC";
+    case SHT_HIPROC: return "SHT_HIPROC";
+    case SHT_LOUSER: return "SHT_LOUSER";
+    case SHT_HIUSER: return "SHT_HIUSER";
+    default:
+        return "UNKNOWN TYPE";
+    }
+}
+std::string Object::print_sections() const {
```

Comment:
This (and the above) can/should be standalone functions in this file (and not declared in the header).